core/src/test/java/org/opensearch/sql/analysis/ExpressionReferenceOptimizerTest.java

```diff
@@ -46,6 +52,29 @@ void group_expression_should_be_replaced() {
     );
   }
 
+  @Test
+  void missing_field_single_field_relevance_query() {
+    assertThrows(SemanticCheckException.class,
+        () -> optimize(dsl.match(
+            dsl.namedArgument("field", DSL.literal("missing_field")),
+            dsl.namedArgument("query", DSL.literal("query_value")))));
+  }
+
+  @Test
+  void missing_field_multi_field_relevance_query() {
```

Comment:
Is it possible to test missing and non missing field in one query?

Reply:
> Should we have integration tests?

Doesn't look like we have a great framework setup for testing `SemanticCheckExceptions` in IT tests. No relevance queries test this so far. I tried a few quick ones but they failed to pass, and I can't test locally with the MLCommons issue.

Reply:
Re IT: have a look on `sqlDeleteSettingsTest`. Can you adopt it for your case?